src/Shaders/spriteMap.fragment.fx

```diff
@@ -0,0 +1,125 @@
+﻿precision highp float;
+
+varying vec3 vPosition;
+varying vec2 vUV;
+varying vec2 tUV;
+
+uniform float time;
+uniform float spriteCount;
+uniform sampler2D spriteSheet;
+uniform vec2 spriteMapSize;
+
+uniform vec2 outputSize;
+uniform vec2 stageSize;
+
+uniform float maxAnimationFrames;
+
+uniform sampler2D frameMap;
+uniform sampler2D tileMaps[LAYERS];
+uniform sampler2D animationMap;
+
+uniform vec3 colorMul;
+uniform float flipU;
+
+#ifdef EDITABLE
+	uniform vec2 mousePosition;
+	uniform float curTile;
+#endif
+
+float mt;
+
+float fdStep = 1./4.;
+mat4 getFrameData(float frameID){
+	float fX = frameID/spriteCount;
+
+	return mat4(
+		texture(frameMap, vec2(fX, 0.), 0.),
+		texture(frameMap, vec2(fX, fdStep*1.), 0.),
+		texture(frameMap, vec2(fX, fdStep*2.), 0.),
+		vec4(0.)
+	);
+}
+
+void main(){
+    vec4 color = vec4(0.);    
+    vec2 tileUV = fract(tUV);
+    if(flipU==1.0){

```

Comment:
We try to avoid using if as much as possible (really slow on low end devices). 
Use #ifdef as much as possible (see other shaders)